### ISSUE TITLE
Simplification of Codeowner Groups

### ISSUE THREAD

**Issue by nstng** [magma/magma#14761]
### Problem

Pull requests are often blocked by missing reviewers. The reason is that maintainer of smaller sub-projects are becoming inactive. This situation is expected to intensify in the transition phase to the upcoming community mode. In order to improve the situation, the codeowner groups should be revised in a way that the community stays actionable and can contribute changes in an acceptable pace. This of course can be revised again later if there is a sufficient number of active contributors for a more detailed codeowner structure.

### Solution

We propose to make the codeownership much simpler by only keeping three groups:
* gateways (combining all groups related to agw, feg, cwf, ...)
* cloud (combining all groups related to orc8r or nms)
* infrastructure (combining all infrastructure-related groups, e.g. ci, docs, bazel)

### Non-goals

This proposal would also benefit from establishing an off boarding workflow for the codeowner groups - see https://github.com/magma/magma/issues/14208. But this effort is not scope of this proposal.

**Comment by nstng** [comment]
Below is a more concrete summary of the proposal above. All existing approver-groups are clustered below the three groups (gateways, cloud, infrastructure). `@magma/approvers-tsc` still remains alone as the tsc approver group.

* @magma/approvers-infra
  * @magma/approvers-ci
  * @magma/approvers-agw-integtests
  * @magma/approvers-docs
  * @magma/approvers-bazel

* @magma/approvers-cloud
  * @magma/approvers-orc8r
  * @magma/approvers-nms
  * @magma/approvers-domain-proxy

* @magma/approvers-gw
  * @magma/approvers-feg
  * @magma/approvers-cwf
  * @magma/approvers-agw
  * @magma/approvers-agw-mme
  * @magma/approvers-agw-containers
  * @magma/approvers-agw-pipelined

* @magma/approvers-tsc

**Comment by nstng** [comment]
Just a note, if this gets accepted - we use some of the groups in manual workflows to check if the person who triggered the workflow is eligible. Search for `approvers-ci` - these places may need to be adapted then.

**Comment by nstng** [comment]
See https://github.com/magma/magma/issues/14822 for the implementation of this proposal.

**Comment by lucasgonze** [comment]
Work completed long ago, but the issue wasn't closed.
